src/erleans_partitions.erl

```diff
@@ -0,0 +1,164 @@
+%%%      or reminders within the interval is to include the Start and End
+%%%      partitions, i.e. Partition >=Start andalso Partition =< End.
+%%% @end
+%%% ---------------------------------------------------------------------------
+-module(erleans_partitions).
+
+-behaviour(gen_server).
+
+-export([start_link/0,
+         find_node/1,
+         add_handler/2,
+         get_range/0]).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-include("erleans.hrl").
+
+-type range() :: {integer(), integer()}.
+
+-record(state, {range          :: range() | undefined,
+                num_partitions :: integer(),
+                node_ranges    :: [{range(), node()}],
+                to_notify      :: #{atom() => pid() | atom()}}).
+
+-define(CH(Item, Partitions), jch:ch(Item, Partitions)).
+
+-spec start_link() -> {ok, pid()} | {error, any()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec find_node(term()) -> {integer(), node()}.
+find_node(Item) ->
+    gen_server:call(?MODULE, {find_node, Item}).
+
+-spec get_range() -> range().
+get_range() ->
+    gen_server:call(?MODULE, get_range).
+
+-spec add_handler(atom(), pid() | atom()) -> ok.
+add_handler(Name, Pid) ->
+    gen_server:call(?MODULE, {add_handler, Name, Pid}).
+
+
+init([]) ->
+    %% callback for changes to cluster membership
+    Self = self(),
+    partisan_peer_service_events:add_sup_callback(fun(Membership) ->
+                                                      Self ! {update, Membership}
+                                                  end),
+    NumPartitions = erleans_config:get(num_partitions),
+    {ok, #state{num_partitions=NumPartitions,
+                node_ranges=[],
+                to_notify=#{}}, 0}.
+
+handle_call({find_node, Item}, From, State=#state{num_partitions=NumPartitions,
+                                                  node_ranges=Ranges}) ->
+    spawn(fun() ->
+              Partition = ?CH(erlang:phash2(Item), NumPartitions),
+              {_, Node} = ec_lists:fetch(fun({{Start, End}, _Node}) when Partition >= Start
+                                                                       , Partition =< End ->
+                                             true;
+                                            (_) ->
+                                             false
+                                         end, Ranges),
+              gen_server:reply(From, {Partition, Node})
+          end),
+    {noreply, State};
+handle_call(get_range, _From, State=#state{range=Range}) ->
+    {reply, Range, State};
+handle_call({add_handler, Name, Pid}, _From, State=#state{to_notify=ToNotify}) ->
+    {reply, ok, State#state{to_notify=ToNotify#{Name => Pid}}}.
+
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info({update, Membership}, State=#state{num_partitions=NumPartitions,
```

Comment:
Yea, hm, we would crash but probably want to match on it. Just not sure what we should do, can simply try to add it back, or can just log and then shutdown, that way if it continues to not be able to work it'll eventually crash the supervisor and bubble up that this node is not working correctly.